blockchain/mutex.go

```diff
@@ -43,9 +42,8 @@ func newMutex() *mutex {
 // Mutex keeps a map of mutex locks per project ID, and it also keeps a track of obtained locks per that ID so it can, after all
 // the locks have been released remove that lock from the mutex map to not pollute memory.
 type mutex struct {
-	mx      *sync.RWMutex               // mutex for access to bellow maps
-	pMutex  map[uuid.UUID]*sync.RWMutex // per project mutexes
-	counter map[uuid.UUID]int           // per project counter
```

Comment:
If we never delete the project locks won't we essentially have memory leaks over time since more and more project IDs will be stored and older ones never get accessed? Maybe we could use another lru cache for this.

Reply:
Yeah this just cleans that since it's not used. I'm aware of that, for now I believe it's not a big deal, but yeah we could add cache but not LRU but time-limited cache with 1 day timeout, because in LRU if you have that many clients locks will be removed. We will revisit this soon. Maybe something you can try working on, but as part of v2